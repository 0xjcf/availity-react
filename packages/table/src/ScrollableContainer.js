import React, { cloneElement } from 'react';
import PropTypes from 'prop-types';

const ScrollableContainer = ({ id, children, ...rest }) => (
  <div id={id} className="av-scrollable-table-wrapper" {...rest}>
    {cloneElement(children, { scrollable: true })}
  </div>
);

ScrollableContainer.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node,
};

export default ScrollableContainer;
