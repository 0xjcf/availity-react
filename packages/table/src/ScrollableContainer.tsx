import React, { cloneElement, isValidElement } from 'react';

type Props = {
  id?: string;
  children?: React.ReactNode | React.ReactChild;
} & React.HTMLAttributes<HTMLElement>;

const ScrollableContainer = ({ id, children, ...rest }: Props): JSX.Element => (
  <div id={id} className="av-scrollable-table-wrapper" {...rest}>
    {isValidElement(children) && children.props.scrollable ? cloneElement(children, { scrollable: true }): children}
  </div>
);

export default ScrollableContainer;
