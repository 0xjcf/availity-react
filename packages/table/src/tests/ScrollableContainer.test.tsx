import React from 'react';
import { render } from '@testing-library/react';
import ScrollableContainer from '../ScrollableContainer';

describe('Scrollable Container', () => {
  test('should render', () => {
    const { container } = render(
      <ScrollableContainer>
        <div>This is my Content.</div>
      </ScrollableContainer>
    );

    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
