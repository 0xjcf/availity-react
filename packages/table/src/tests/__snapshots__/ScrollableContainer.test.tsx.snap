// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollable Container should render 1`] = `
<div>
  <div
    class="av-scrollable-table-wrapper"
  >
    <div
      scrollable="true"
    >
      This is my Content.
    </div>
  </div>
</div>
`;

exports[`Scrollable Container should render multiple children 1`] = `
<div>
  <div
    class="av-scrollable-table-wrapper"
  >
    <div
      scrollable="true"
    >
      This is my Content.
    </div>
    <div
      scrollable="true"
    >
      This is my Content.
    </div>
  </div>
</div>
`;
